### PR TITLE
Migrate documentation deployment from Travis CI to GitHub Actions

### LINKED ISSUE #566: Migrate Travis CI jobs to GitHub Actions
**Description of the desired feature**

We recently migrated all the CI testings and style checks from Azure Pipelines to GitHub Actions. Currently, we're using both GitHub Actions and Travis CI.

The Travis CI is doing following jobs:

- ~~Test PyGMT with Python 3.6-3.8 on Linux~~
- [ ] Deploy documentation to gh-pages branch for master branch and tags
- [x] Publish PyGMT to PyPi for tags (#679)

The first one is already covered in the GitHub Actions, so it's OK to just remove them from Travis CI.

The 2nd and the 3rd one should be migrated to GitHub Actions.

**Are you willing to help implement and maintain this feature?** Yes, but contributions are welcome.

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This is the last step of the CI migration from Travis to GitHub Actions. 

The codes are copied from https://github.com/fatiando/boule/blob/master/.github/workflows/docs.yml.

For testing purpose, the deployment steps are enabled for PRs temporarily, and the documentation is deployed to the `gh-pages-testing` branch.

It seems working very well (see https://github.com/GenericMappingTools/pygmt/tree/gh-pages-testing).

Fixes #566.

**TODO**

- [x] Revert the changes in commit 4ed4de6932baf127ec71efc6c304b24bd4f28b94
- [x] Remove `.travis.yml`
- [x] Remove the Travis badge in README
- [x] Update CONTRIBUTING.md
- [x] Update MAINTENANCE.md

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
   push:
     branches: [ master ]
   pull_request:
+  release:
+    types:
+      - published
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:12]: Also trigger builds for tags/releases.

File: .github/workflows/ci_tests.yaml [hunk 2]
```diff
@@ -138,3 +141,66 @@ jobs:
           file: ./coverage.xml # optional
           env_vars: OS,PYTHON
           fail_ci_if_error: true
+
+      - name: Checkout the gh-pages-testing branch
+        uses: actions/checkout@28c7f3d2b5162b5ddd3dfd9a45aa55eaf396478b
+        with:
+          ref: gh-pages-testing
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:148]: Use `gh-pages-testing` branch for testing purpose. Need to revert the branch name to `gh-pages` before merging.

File: .github/workflows/ci_tests.yaml [hunk 3]
```diff
@@ -138,3 +141,66 @@ jobs:
...
+          # Checkout to this folder instead of the current one
+          path: deploy
+          # Download the entire history
+          fetch-depth: 0
+        if: (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
+        #if: (github.event_name == 'release' || github.event_name == 'push') && (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:154]: The `if` condition is a little complicated here.

We don't want to deploy documentation for PRs, thus we limit the triggering event to `release` or `push`.

We don't want to deploy the same documentation multiple times, so we check if we're using Ubuntu + Python 3.9. 

Need to use the commented `if` condition before merging.

File: .github/workflows/ci_tests.yaml [hunk 4]
```diff
@@ -138,3 +141,66 @@ jobs:
...
+              echo -e "\nMaking a new commit:"
+              git commit -m "$message"
+          fi
+          # Make the push quiet just in case there is anything that could leak
+          # sensitive information.
+          echo -e "\nPushing changes to gh-pages-testing."
+          git push -fq origin gh-pages-testing 2>&1 >/dev/null
+          echo -e "\nFinished uploading generated files."
+        if: (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
+        #if: (github.event_name == 'release' || github.event_name == 'push') && (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:206]: The same as above.

**Comment by seisman** [comment]
Ping @GenericMappingTools/python for review.

**Review by weiji14 — APPROVED** [review]
Sorry for the delay in review. A quick scan and this seems fine to me, though I would prefer if @leouieda (who has more experience on Github Actions docs deploy) take another look. Otherwise, I think you can make the necessary pre-merge changes mentioned at https://github.com/GenericMappingTools/pygmt/pull/713/files#r535384202 and other places.

**Pull request merged**